src/DataViews/DataView.cpp

```diff
@@ -221,24 +221,55 @@ void DataView::OnUnselectRequested(const std::vector<int>& selection) {
 }
 
 void DataView::OnEnableFrameTrackRequested(const std::vector<int>& selection) {
+  metrics_uploader_->SendLogEvent(
+      orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_ENABLE_CLICKED);
+
   for (int i : selection) {
     const FunctionInfo& function = *GetFunctionInfoFromRow(i);
     // Functions used as frame tracks must be hooked (selected), otherwise the
     // data to produce the frame track will not be captured.
     app_->SelectFunction(function);
+
     app_->EnableFrameTrack(function);
-    app_->AddFrameTrack(function);
+    metrics_uploader_->SendLogEvent(
+        orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_ENABLED);
+
+    switch (app_->AddFrameTrack(function)) {
+      case AppInterface::AddFrameTrackResult::kNotPossible:
+        break;
+      case AppInterface::AddFrameTrackResult::kVisible:
+        metrics_uploader_->SendLogEvent(
+            orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_ADDED_VISIBLE);
+        break;
+      case AppInterface::AddFrameTrackResult::kInvisible:
+        metrics_uploader_->SendLogEvent(
+            orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_ADDED_INVISIBLE);
+        break;
+    }
   }
 }
 
 void DataView::OnDisableFrameTrackRequested(const std::vector<int>& selection) {
+  metrics_uploader_->SendLogEvent(
+      orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_DISABLE_CLICKED);
+
   for (int i : selection) {
     const FunctionInfo& function = *GetFunctionInfoFromRow(i);
     // When we remove a frame track, we do not unhook (deselect) the function as
     // it may have been selected manually (not as part of adding a frame track).
     // However, disable the frame track, so it is not recreated on the next capture.
     app_->DisableFrameTrack(function);
-    app_->RemoveFrameTrack(function);
+    metrics_uploader_->SendLogEvent(
+        orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_DISABLED);
+
+    switch (app_->RemoveFrameTrack(function)) {
+      case AppInterface::RemoveFrameTrackResult::kNotPossible:
+        break;
+      case AppInterface::RemoveFrameTrackResult::kSuccess:
+        metrics_uploader_->SendLogEvent(
+            orbit_metrics_uploader::OrbitLogEvent::ORBIT_FRAME_TRACK_REMOVED);
+        break;
+    }
```

Comment:
Same here. I think it might be better suited to do that in App::RemoveFrameTrack. It makes the metric also work if we call `RemoveFrameTrack from some other place in the future.

Reply:
Done! Thanks!